File: backend/app/models/conversation.py
"""会话持久化模型。

conversations：一轮轮问答归属的会话（title 取首轮提问前 40 字）。
chat_messages：会话内的 user/assistant 消息，按 id 正序即时间正序。

历史会话抽屉与"继续对话"依赖这两张表（phase14）。
"""

from __future__ import annotations

import uuid
from datetime import datetime

from sqlalchemy import DateTime, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column

from app.models.user import Base


def _gen_uuid() -> str:
    return str(uuid.uuid4())


class Conversation(Base):
    """会话表。"""

    __tablename__ = "conversations"

    id: Mapped[str] = mapped_column(String(36), primary_key=True, default=_gen_uuid)
    user_id: Mapped[str] = mapped_column(String(64), index=True, nullable=False)
    title: Mapped[str] = mapped_column(String(120), default="")
    created_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, server_default=func.now(), onupdate=func.now()
    )


class ChatMessage(Base):
    """会话消息表（role: user | assistant）。"""

    __tablename__ = "chat_messages"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    conversation_id: Mapped[str] = mapped_column(String(36), index=True, nullable=False)
    role: Mapped[str] = mapped_column(String(16), nullable=False)
    content: Mapped[str] = mapped_column(Text, nullable=False)
    intent: Mapped[str] = mapped_column(String(16), default="")
    created_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now())

File: frontend/src/api/conversation.js
import request from './request'

// 会话列表：GET /api/v1/conversations?limit=&offset=
// 返回 {conversations: [{id, title, created_at, updated_at, message_count}], total}
// （仅本人会话，updated_at 倒序）
export function listConversations(limit = 20, offset = 0) {
  return request.get('/conversations', { params: { limit, offset } })
}

// 会话历史消息：GET /api/v1/conversations/{id}/messages
// 返回 {conversation_id, messages: [{role: 'user'|'assistant', content, intent, created_at}]}
// （按时间正序；他人会话或不存在统一 404 防枚举）
export function listMessages(conversationId) {
  return request.get(`/conversations/${conversationId}/messages`)
}
